chainer/functions/math/matmul.py

```diff
@@ -181,11 +175,9 @@ def __init__(self, transa=False, transb=False):
         self.transb = transb
 
     def _output_shape(self, a, b):
-        batch_size = a.shape[0]
-        a_mat_shape = _as_mat(a[0]).shape
-        b_mat_shape = _as_mat(b[0]).shape
-        m = a_mat_shape[1 if self.transa else 0]
-        n = b_mat_shape[0 if self.transb else 1]
+        batch_size = len(a)
+        m = _as_batch_mat(a).shape[2 if self.transa else 1]
```

Comment:
We do not have to do reshape `a` itself. We just need the shape of reshaped `a`.